lib/config/configuration.go

```diff
@@ -686,6 +686,34 @@ func applyProxyConfig(fc *FileConfig, cfg *service.Config) error {
 		}
 		cfg.Proxy.TunnelPublicAddrs = addrs
 	}
+	if len(fc.Proxy.PostgresPublicAddr) != 0 {
+		// Postgres proxy is multiplexed on the web proxy port. If the port is
+		// not specified here explicitly, prefer defaults in the following
+		// order, depending on what's set:
+		//   1. Web proxy public port
+		//   2. Web proxy listen port
+		//   3. Web proxy default listen port
+		defaultPort := cfg.Proxy.WebAddr.Port(defaults.HTTPListenPort)
+		if len(cfg.Proxy.PublicAddrs) != 0 {
+			defaultPort = cfg.Proxy.PublicAddrs[0].Port(defaults.HTTPListenPort)
```

Comment:
```suggestion
			defaultPort = cfg.Proxy.PublicAddrs[0].Port(defaultPort)
```

Reply:
This is on purpose actually. If there is no port in public address, we default to the "default" respective port, not listen address port. This is consistent with how default ports for other public addresses are handled above.